### ISSUE TITLE
assocPath breaks if path includes digits

### ISSUE THREAD

**Issue by aethelz** [selfrefactor/rambda#524]
assocPath breaks if string path starts with digit.
It seems that [this line](https://github.com/selfrefactor/rambda/blob/65b780f236595c34677726b2370e68cec0031f9b/src/assocPath.js#L22) is to blame.

Minimal reproducible example (comparison with correct ramda behavior included): https://stackblitz.com/edit/js-e5ro2z?file=index.js

**Comment by selfrefactor** [comment]
Thank you for the report - it is a valid bug and I will try to apply a fix tomorrow and make a new release.

**Comment by selfrefactor** [comment]
Fixed with `5.12.1`

**Comment by selfrefactor** [comment]
@aethelz - I am reopening this as the change I made to fix it, caused `R.lensPath` to stop working with strings as path input.

Now I restored and changed the lines in question and we have this bug tests:

```

test('bug', () => {
  /*
    https://github.com/selfrefactor/rambda/issues/524
  */
  const state = {}

  const result = assocPath(
    [ 'outerProp', '2020-03-10' ], { prop : 2 }, state
  )
  const incorrect = assocPath(
    [ 'outerProp', '5' ], { prop : 2 }, state
  )

  const expected = { outerProp : { '2020-03-10' : { prop : 2 } } }
  const incorrectExpected = {
    outerProp : [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      { prop : 2 },
    ],
  }
  expect(result).toEqual(expected)
  expect(incorrect).toEqual(incorrectExpected)
})
```

It is not the best solution, as it creates a difference with Ramda API. In other words, using numbers as path in `R.assocPath` is problematic. I will try to suggest better solution, while I wait for your feedback.

**Comment by selfrefactor** [comment]
Properly fixed with `5.13.0` as `5.12.1` cause `R.lensPath` to stop working with string as path input.
